src/arch/probe.cc

```diff
@@ -15,7 +15,7 @@ int ceph_arch_probe(void)
   ceph_arch_intel_probe();
 #elif defined(__arm__) || defined(__aarch64__)
   ceph_arch_arm_probe();
-#elif defined(__powerpc__) || defined(__ppc__)
+#elif defined(HAVE_POWER8)
```

Comment:
@ellert hi Mattias, i think #42962 is a better fix.

because we check for the architecture using the predefined macros offered by the compiler, like `__i386__` and `__arm__` in this source file. would be more consistent if we can use `__powerpc__` to detect PowerPC architectures as well.

what do you think?

Reply:
This is what is done before this proposed change, and it does not work.
That it doesn't work is the reason for proposing this change.
I tried to explain why in the description and the commit message, but I will expand.

In the file src/arch/CMakeLists.txt

https://github.com/ceph/ceph/blob/0917e204e9367cd31dbf25884fb32c74cb2e775a/src/arch/CMakeLists.txt#L8

The HAVE_POWER8 conditional is used to instruct cmake to compile or not compile the code in src/arch/ppc.c.

This means that if you try to call the functions or access variables defined in src/arch/ppc.c when HAVE_POWER8 is not true, that will fail since they do not exist then. The compilation fails with:
```
/usr/bin/ld: ../lib/libceph-common.so.0: undefined reference to `ceph_arch_ppc_probe'
/usr/bin/ld: ../lib/libceph-common.so.0: undefined reference to `ceph_arch_ppc_crc32'
/usr/bin/ld: ../lib/libceph-common.so.0: undefined reference to `ceph_crc32c_ppc'
collect2: error: ld returned 1 exit status
```
The condition
```if defined(__powerpc__) || defined(__ppc__)```
does not guarantee that
```if defined(HAVE_POWER8)```
is true.

Reply:
Sorry, I missed the first line of your reply. So my previous comment was not relevent. I deleted it.
Your suggestion looks OK, I will try to build with your patch and get back to you.

Reply:
@ellert i am not sure if you've read #42962. if not, could you take a look?

Reply:
> The compilation on Debian 32 bit powerpc succeeded with your changes.

@ellert closing this PR in favor of https://github.com/ceph/ceph/pull/42962